lib/figaro.rb

```diff
@@ -0,0 +1,62 @@
+require 'yaml'
+require 'erb'
+
+class Figaro
+  def self.env
+    Environment
+  end
+
+  def self.setup(path)
+    Environment.setup(path)
+  end
+
+  def self.require_keys(keys)
+    Environment.require_keys(keys)
+  end
+
+  private
+end
+
+class Environment
+  def self.setup(path, env = Rails.env)
+    @config = {}
+    values = YAML.load(ERB.new(File.read(path)).result)
+
+    keys = Set.new(values.keys - ['production', 'development', 'test'])
+    keys |= values[env].keys
+
+    keys.each do |key|
+      value = values[env][key] || values[key]
+
+      if key == key.upcase
+        ENV[key] = value
+      end
+
+      @config[key] = value
+    end
+
+    @config
+  end
+
+  def self.method_missing(m, *args, &block)
```

Comment:
Can we also implement `respond_to_missing?`? Good ruby practice is to implement that every time we add `method_missing`, here's the first blog post I found that agreed with me: https://thoughtbot.com/blog/always-define-respond-to-missing-when-overriding